onnx/utils.py

```diff
@@ -79,6 +79,14 @@ def _collect_reachable_nodes(
         nodes = [n for n in self.graph.node if n in reachable_nodes]
         return nodes
 
+    def _collect_referred_local_functions(
+            self,
+            nodes,  # type: List[NodeProto]
+    ):  # type: (...) -> List[FunctionProto]
+        node_op_types = {n.op_type for n in nodes}
+        local_functions = [f for f in self.model.functions if f.name in node_op_types]
+        return local_functions
```

Comment:
I think we may need a loop to accumulate functions called from other functions. Main-graph may call local function foo1, and function foo1 may contain a call to local function foo2, and so on. We should collect foo2 also.

Reply:
good point. coded for this case